### ISSUE TITLE
Add support for BNO055 9-DOF IMU

### ISSUE THREAD

**Issue by thall4** [usfsoar/NSL_18-19_Rover#11]
Needs to be able to detect if we run into something and stop moving, and then adjust course accordingly. Doesn't have to be anything fancy, like:
```c++
if (x_accel < 0 && motor_direction == forward)
reverse();
delay(500);
leftturn();
delay(200);
forward();
//normal operation follows
```

**Comment by iansan5653** [comment]
The new IMU you ordered is a 9-axis. Is the 345 onboard?

**Comment by iansan5653** [comment]
We used an ADXL345 in the airbrakes. Relevant code:
Setup:
https://github.com/usfsoar/NSL_18-19_Airbrakes/blob/master/main.ino#L64-L69

Usage (about as simple as it gets; we just pulled y-accel):
https://github.com/usfsoar/NSL_18-19_Airbrakes/blob/master/main.ino#L192-L195

**Comment by thall4** [comment]
I filled this out before I decided to order another sensor, the new one I ordered has an embedded gyro which will make it more useful probably.

**Comment by iansan5653** [comment]
Also requires
```
#include <Adafruit_ADXL345_U.h>
#include <Adafruit_Sensor.h>
```

**Comment by iansan5653** [comment]
Yeah the new one should support this feature as well as compass. What's the new one called?

**Comment by thall4** [comment]
9-DOF Absolute Orientation IMU Fusion Breakout - BNO055

**Comment by iansan5653** [comment]
`accel < 0` is not a good check btw; they would trigger any time the rover decelerates, ie, if it's climbing a hill.  Not sure what the new IMU supports but some sort of significant spike in negative acceleration would be better (unless the IMU supports something better I mean).

**Comment by thall4** [comment]
Yeah I know, that was me just spitballing a basic trigger off the top of my head

**Comment by iansan5653** [comment]
Started a branch `feature/imu` with a draft pull request so we can add the feature over time with multiple people working on it.
